pkg/operatorsource/appregistryhelper.go

```diff
@@ -5,6 +5,7 @@ import (
 
 	marketplace "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
 	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+	interface_client "github.com/operator-framework/operator-marketplace/pkg/client"
```

Comment:
If combining the phases fixes the upgrade issue, we would also need to backport #147 to 4.1.z. Is this desired?

Reply:
@kevinrizza any thoughts?

Reply:
@aravindhp thanks for noticing this - it completely escaped me! Will it be harder/riskier to get this PR in 4.1.z if it includes #147 ? I am ok with applying the changes for combining the phases on top of the 4.1 changes and re-writing the unit tests to minimize potential bugs and accelerate the merge process. I can get that done this afternoon. What do you think?

Reply:
I was looking at the code and given #147 has changed configuring.go it will be hard to separate the two. We might need to fix this separately for 4.1.z if needed. Lets leave it as it is for now.